File: internal/core/domain/ports/payment_repository.go
package ports

import (
	"context"

	"github.com/fiap/ms-payment/internal/core/domain/entities"
)

type PaymentRepository interface {
	GetByExternalId(ctx context.Context, externalId string) (entities.Payment, error)
	Create(ctx context.Context, payment entities.Payment) error
	Delete(ctx context.Context, externalId string) error
}

File: src/core/domain/ports/payment_repository.ts
import { Payment } from "@entities/payment"

export interface IPaymentRepository {
	getByExternalId(externalId: string): Promise<Payment | null>
	create(payment: Payment): Promise<void>
	delete(externalId: string): Promise<void>
	updateStatusByExternalId(externalId: string, status: string): Promise<void>
}
